k8s/test/e2e/distributed-assembly-job3.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020-2022 Alibaba Group Holding Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: distributed-assembly-job3-deployment
+  namespace: vineyard-job
+spec:
+  selector:
+    matchLabels:
+      app: distributed-assembly-job3
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        # The label is limited to 63 characters, so we need to use the annotation here
+        scheduling.k8s.v6d.io/required: distributed-assembly-job1.distributed-assembly-job2
+      labels:
+        app: distributed-assembly-job3
+        assembly.v6d.io/enabled: "true"
+        # this label represents the vineyardd's name that need to be used
+        scheduling.k8s.v6d.io/vineyardd: vineyardd-sample
+        scheduling.k8s.v6d.io/job: distributed-assembly-job3
+    spec:
+      schedulerName: vineyard-scheduler
+      containers:
+        - name: job3
+          image: docker.pkg.github.com/v6d-io/v6d/distributed-assembly-job3
+          imagePullPolicy: IfNotPresent
+          # TODO: could we make it more native?
+          env:
+            - name: JOB_NAME
+              value: distributed-assembly-job3
+            - name: REQUIRED_JOB_NAME
+              value: distributed-assembly-job2
+          envFrom:
+          - configMapRef:
+              name: distributed-assembly-job2
+          volumeMounts:
+            - mountPath: /var/run
+              name: vineyard-sock
+      volumes:
+        - name: vineyard-sock
+          hostPath:
+            path: /var/run/vineyard-vineyard-system-vineyardd-sample
```

Comment:
The files `k8s/test/e2e/distributed-assembly-job{1,2,3}.yaml` looks exactly same, and only differ at certain configuration entries.

Could we use it as a "template" and render those arguments in the usage site using Kustomize or something like that?

Reply:
Yeah, thanks for your advice, now the multiple yamls are combined in a general yaml,  then we could use `gomplate` and `sed` to create different job yamls.